node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -84,17 +91,38 @@ impl<C: Context> ActiveValidator<C> {
         start_time: Timestamp,
         state: &State<C>,
     ) -> (Self, Vec<Effect<C>>) {
+        let own_last_unit = Self::read_last_unit(state.params().unit_hash_file());
         let mut av = ActiveValidator {
             vidx,
             secret,
             next_round_exp: state.params().init_round_exp(),
             next_timer: state.params().start_timestamp(),
             next_proposal: None,
+            own_last_unit,
         };
         let effects = av.schedule_timer(start_time, state);
         (av, effects)
     }
 
+    fn read_last_unit<P: AsRef<Path>>(path: P) -> Option<C::Hash> {
+        let mut file = File::open(path).ok()?;
```

Comment:
Will this `.ok()?` exit early only if the file does not exist? I am wondering whether the cases like a corrupted file or inability to open it should be panics instead.

Reply:
Ah, but it's a "hot fix"…

Reply:
Fixed it :+1: